src/core/Toast/Toast.tsx

```diff
@@ -132,20 +163,50 @@ export const Toast = (props: ToastProps) => {
     }
   };
 
+  const calculateOutAnimation = (node: HTMLElement) => {
+    // calculation translate x and y pixels.
+    let translateX = 0;
+    let translateY = 0;
+    if (animateOutTo && node) {
+      const { x: startX, y: startY } = node.getBoundingClientRect(); // current element
+      const { x: endX, y: endY } = animateOutTo.getBoundingClientRect(); // anchor point
+      translateX = endX - startX;
+      translateY = endY - startY;
+    }
+    return { translateX: translateX, translateY: translateY };
+  };
+
   return (
-    <Transition
-      timeout={{ enter: 240, exit: 120 }}
+    <CSSTransition
+      timeout={{ enter: 240, exit: animateOutTo ? 400 : 120 }}
       in={visible}
       appear={true}
       unmountOnExit={true}
       onExited={onRemove}
+      onEnter={(node) => {
+        node.style.transform = 'translateY(15%)';
+        node.style.transitionTimingFunction = 'ease';
+      }}
+      onEntered={(node) => {
+        node.style.transform = 'translateY(0)';
+      }}
+      onExiting={(node) => {
+        const vars = calculateOutAnimation(node);
```

Comment:
deconstruct straight away, cause `vars` is not the best name here.
`const {translateX, translateY} = calculateOutAnimation(node)`